[Flaky test] TestTraceExporter times out on CI

### Component(s)

exporter/datadog

### What happened?

Example: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6485792155/job/17612673255

```
--- FAIL: TestTraceExporter (5.01s)
    traces_exporter_test.go:277: Timed out
FAIL
```

Cannot reproduce the failure with my local test run.

### Collector version

mainline

### Environment information

## Environment

go version go1.21.1 darwin/arm64


### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/exporter/datadogexporter/traces_exporter_test.go
+++ b/exporter/datadogexporter/traces_exporter_test.go
@@ -229,7 +229,6 @@ func TestTracesSource(t *testing.T) {
 }
 
 func TestTraceExporter(t *testing.T) {
-	t.Skip("Flaky test on CI, see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27630")
 	metricsServer := testutil.DatadogServerMock()
 	defer metricsServer.Close()
 
@@ -259,6 +258,7 @@ func TestTraceExporter(t *testing.T) {
 			},
 			IgnoreResources: []string{},
 			flushInterval:   0.1,
+			TraceBuffer:     2,
 		},
 	}
 

